Cargo.toml

```diff
@@ -32,7 +32,7 @@ tokio = { version = "1", features = ["full"] }
 itertools = "0.10.5"
 once_cell = "1.16.0"
 thiserror = "1"
-
+machine-uid = "0.5.1"
```

Comment:
Cool crate!

> In Linux, machine id is a single newline-terminated, hexadecimal, 32-character, lowercase ID. When decoded from hexadecimal, this corresponds to a 16-byte/128-bit value. This ID may not be all zeros. This ID uniquely identifies the host. It should be considered “confidential”, and must not be exposed in untrusted environments. And do note that the machine id can be re-generated by root.

Should we hash the id for privacy reasons?

Reply:
So unfortunately it doesn't take darwin architecture into account. I'll make a PR on their crate but don't knwo when we should be able to use it for ARM users

Reply:
@Rizary might be interesting to you, too

Reply:
https://github.com/ARK-Builders/arklib/issues/61